server/src/main/java/io/crate/planner/Planner.java

```diff
@@ -153,13 +156,15 @@
 import io.crate.user.UserManager;
 
 @Singleton
-public class Planner extends AnalyzedStatementVisitor<PlannerContext, Plan> {
+public class Planner {
 
     private static final Logger LOGGER = LogManager.getLogger(Planner.class);
 
     private final ClusterService clusterService;
     private final TableStats tableStats;
     private final LogicalPlanner logicalPlanner;
+    private final OptimizerProgressTracker defaultTracker = LoggingOptimizerProgressTracker.getInstance();
```

Comment:
This should probably get called at runtime and not be initialized eagerly, otherwise the `LOGGER.isTraceEnabled()` check in `getInstance` will never observe runtime log level changes.

Reply:
Now it is called each time a new `PlannerContext` is created.